contracts/Remittance.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity 0.5.10;
+
+import "./SafeMath.sol";
+import "./Stoppable.sol";
+
+contract Remittance is Stoppable {
+    using SafeMath for uint;
+    
+    struct Order {
+        uint time;
+        address emitter;
+        uint amount;
+    }
+    
+    uint constant TX_FEES = 2000;
+
+    mapping(bytes32 => Order) public transactions;
+    mapping(address => uint) public fees;
+    
+    event LogNewTransaction(address indexed emitter, address exchange, uint amount);
+    event LogTakeFees(address indexed emitter, uint amount);
+    event LogCancelTransaction(address indexed emitter, address exchange, uint amount);
+    event LogWithdraw(address indexed emitter, uint amount);
+    
+    function hashOTP(address exchange, bytes32 password) public view returns(bytes32 hash) {
+        require(exchange != address(0) && password != 0, "Both exchange and password must be set");
+        hash = keccak256(abi.encodePacked(address(this), exchange, password));
+    }
+    
+    function newTransaction(address exchange, bytes32 hashedOTP, uint delay) _onlyIfRunning public payable returns(bool success) {
+        require(exchange != address(0) && hashedOTP != 0 && msg.value > 0, "An error occured. Ensure that both exchange and the secret are set and that your transaction value is above 0");
+        require(delay > 0 days, "Delay should be above 0 day");
+        require(transactions[hashedOTP].emitter == address(0), "Password already used");
+        
+        uint amount = msg.value.sub(TX_FEES);
+
+        fees[getOwner()] = fees[getOwner()].add(TX_FEES);
+        emit LogTakeFees(msg.sender, TX_FEES);
+        
+        emit LogNewTransaction(msg.sender, exchange, amount);
```

Comment:
You notice that `exchange` here is purely informative. There is no enforcement to ensure its validity.
Perhaps rename it to express that?

Reply:
Agree, the exchange is purely informative however, its value is checked ;). cf. Line 31